tests/ltp/patches/fix_setresuid_setresuid04.patch

```diff
@@ -0,0 +1,214 @@
+commit 610698772ff5d79b7592aea1b8f78757a11a68f0
+Author: shaik shavali <shaik.shavali1@wipro.com>
+Date:   Wed May 6 04:24:46 2020 +0000
+
+    This patch is nesesary to fix test failure in sgxlkl
+    environment and to test the syscall "setresuid"
+    behaves correctly with file permissions in
+    multithreaded environment.
+
+diff --git a/testcases/kernel/syscalls/setresuid/setresuid04.c b/testcases/kernel/syscalls/setresuid/setresuid04.c
+index e197476ff..fceafa441 100644
+--- a/testcases/kernel/syscalls/setresuid/setresuid04.c
++++ b/testcases/kernel/syscalls/setresuid/setresuid04.c
+@@ -25,8 +25,8 @@
+  * 	Check if setresuid behaves correctly with file permissions.
+  *      The test creates a file as ROOT with permissions 0644, does a setresuid
+  *      and then tries to open the file with RDWR permissions.
+- *      The same test is done in a fork to check if new UIDs are correctly
+- *      passed to the son.
++ *      The same test is done in a thread to check if process UIDs correctly
++ *      passed to new thread.
+  *
+  * USAGE:  <for command-line>
+  *  setresuid04 [-c n] [-e] [-i n] [-I x] [-P x] [-t]
+@@ -54,6 +54,7 @@
+ #include "safe_macros.h"
+ #include <pwd.h>
+ #include "compat_16.h"
++#include <pthread.h>
+ 
+ TCID_DEFINE(setresuid04);
+ int TST_TOTAL = 1;
+@@ -65,46 +66,68 @@ int fd = -1;
+ 
+ void setup(void);
+ void cleanup(void);
+-void do_master_child();
++void* do_master_thread(void*);
++void* do_sub_thread(void*);
+ 
+ int main(int ac, char **av)
+ {
+-	pid_t pid;
++	pthread_t threadid;
+ 
+ 	tst_parse_opts(ac, av, NULL, NULL);
+ 	setup();
+ 
+-	pid = FORK_OR_VFORK();
+-	if (pid < 0)
+-		tst_brkm(TBROK, cleanup, "Fork failed");
+-
+-	if (pid == 0)
+-		do_master_child();
++	//create a new thread to perform the test
++	if(pthread_create(&threadid, NULL, do_master_thread, NULL))
++	{
++		tst_brkm(TBROK, NULL, "TEST_FAILED: Thread creation failed");
++		tst_exit();
```

Comment:
Why is `tst_exit();` required here? The original code doesn't call it. Doesn't `tst_brkm` exit already?

Reply:
I will modify and update the patch